models/exchange/binance/api.py

```diff
@@ -65,7 +67,17 @@ def __init__(self, api_key: str='', api_secret: str='', api_url: str='https://ap
         self.api_url = api_url
         self.api_key = api_key
         self.api_secret = api_secret
-        self.client = Client(self.api_key, self.api_secret, { 'verify': False, 'timeout': 20 })
+        
+        for i in range(10):
+            try:
+                self.client = Client(self.api_key, self.api_secret, { 'verify': False, 'timeout': 20 })
+                break
+            except:
+                if i == 9:
+                    raise SystemExit("Can not create instance of AuthAPI client.")
+                print('Exception: ', sys.exc_info()[0]) 
+                print('Error on creating instance of AuthAPI Client. Trying again... Attempt: ', i)
+                sleep(0.1)
```

Comment:
0.1 is very short. Would it not be better to set say 5-10 seconds just to give it a chance to respond?

Reply:
The logs below are shows that problem is not about to first connection to Exchange.
Its about to API request limit of the Binance.

If we put 5-10 second waiting there we will be stopping bot for that long.
I implemented and tested this and what I see if we catch exception;
most of the time we get the Client() object instance at first try.

Reply:
Also the code is duplicated in `AuthAPI` and `PublicAPI` L329-340, to make maintenance easier, why to not use the inherited class `AuthAPIBase` just adding a method to get the client instance and after use it in both classes that already inherit this class ?

Reply:
You are correct, this just a little bit saving the day,
We should not need to connecting APIs for every another call.
Auth can be handled in Public API constructor and PublicAPI can be a global variable, or part of something like state.
So we would not create an API conncetion for every request.